code/src/LeafletProvider/DrawingTools/DrawMarker.ts

```diff
@@ -0,0 +1,143 @@
+/// <reference path="AbstractProviderTool.ts" />
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+namespace LeafletProvider.DrawingTools {
+    export class DrawMarker extends AbstractProviderTool<Configuration.DrawingTools.DrawMarkerConfig> {
+        private _defaultIcon: L.DivIcon;
+
+        constructor(
+            map: OSFramework.OSMap.IMap,
+            drawingTools: OSFramework.DrawingTools.IDrawingTools,
+            drawingToolsId: string,
+            type: string,
+            configs: Configuration.DrawingTools.DrawMarkerConfig
+        ) {
+            super(
+                map,
+                drawingTools,
+                drawingToolsId,
+                type,
+                new Configuration.DrawingTools.DrawMarkerConfig(configs)
+            );
+
+            this._defaultIcon = new L.DivIcon({
+                iconSize: [24, 40],
+                className: 'marker-leaflet-icon',
+                iconAnchor: [12, 40]
+            });
+        }
+
+        /**
+         * Sets the new icon on the Marker just by using the iconUrl
+         * The width and the height of the icon will use the one in the configs (if set) or will use the default image size
+         * The icon will be centered by x axis on the marker position but on the y axis it will appear right above it
+         */
+        private _createIcon(iconUrl: string): L.DivIcon {
+            let icon: L.DivIcon;
+            // If the iconUrl is not set or is empty, we should use the defaultIcon
+            if (iconUrl === '') {
+                // icon will have the following configurations:
+                // iconSize: [24, 40],
+                // className: 'marker-leaflet-icon',
+                // iconAnchor: [12, 40]
+                icon = this._defaultIcon;
+            } else {
+                let iconSize: L.PointExpression;
+                let iconAnchor: L.PointExpression; /*
+                // If the size of the icon is defined by a valid width and height, use those values
+                // Else If nothing is passed or the icon size has the width or the height equal to 0, use the full image size
+                if (this.config.iconWidth > 0 && this.config.iconHeight > 0) {
+                    iconSize = [this.config.iconWidth, this.config.iconHeight];
+                    // The icon will be centered by x axis on the marker position but on the y axis it will appear right above it
+                    iconAnchor = [
+                        this.config.iconWidth / 2,
+                        this.config.iconHeight
+                    ];
+                }*/
```

Comment:
Having this comment here makes sense because we might want to have this in the future, but it would be good to have a TODO mark here